pkg/autoscaler/autoscaler.go

```diff
@@ -175,6 +175,13 @@ func (a *Autoscaler) Record(ctx context.Context, stat Stat) {
 func (a *Autoscaler) Scale(ctx context.Context, now time.Time) (int32, bool) {
 	logger := logging.FromContext(ctx)
 
+	a.specMux.RLock()
+	stableWindow := a.deciderSpec.MetricSpec.StableWindow
+	panicWindow := a.deciderSpec.MetricSpec.PanicWindow
+	target := a.deciderSpec.TargetConcurrency
+	panicThreshold := a.deciderSpec.PanicThreshold
+	a.specMux.RUnlock()
```

Comment:
Would it make this slightly more readable if we have a `currentConfig` method or something that returns all of these?

```golang
func (a *Autoscaler) currentConfig() (stableWindow, panicWindow, target, panicThreshold)
```

Or even easier, return a deepcopy of the DeciderSpec (similar to how we used to return a deepcopy of the configuration)

Reply:
We could just return the `DeciderSpec` without deep copy, since it's all by value.